src/bootstrap-table.js

```diff
@@ -697,7 +697,7 @@
             return;
         }
 
-        this.fromHtml = true;
+        this.fromHtml = this.options.sidePagination === 'server' ? false : true;
```

Comment:
`this.fromHtml = this.options.sidePagination === 'client';`
or `this.fromHtml = this.options.sidePagination !== 'server';`